OrbitCore/Path.h

```diff
@@ -10,48 +10,47 @@
 
 class Path {
  public:
-  static void Init();
-
-  static std::string GetExecutableName();
-  static std::string GetExecutablePath();
-  static std::string GetBasePath();
-  static std::string GetDllPath(bool a_Is64Bit);
-  static std::string GetDllName(bool a_Is64Bit);
-  static std::string GetFileMappingFileName();
-  static std::string GetSymbolsFileName();
-  static std::string GetCachePath();
-  static std::string GetPresetPath();
-  static std::string GetPluginPath();
-  static std::string GetCapturePath();
-  static std::string GetDumpPath();
-  static std::string GetAppDataPath();
-  static std::string GetLogFilePath();
-  static std::string GetIconsPath();
-  static void Dump();
+  [[nodiscard]] static std::string GetExecutablePath();
+  [[nodiscard]] static std::string GetExecutableDir();
+  [[nodiscard]] static std::string GetBasePath();
+  [[nodiscard]] static std::string GetDllPath(bool a_Is64Bit);
+  [[nodiscard]] static std::string GetDllName(bool a_Is64Bit);
+  [[nodiscard]] static std::string GetFileMappingFileName();
+  [[nodiscard]] static std::string GetSymbolsFileName();
+  [[nodiscard]] static std::string CreateAndGetCacheDir();
+  [[nodiscard]] static std::string CreateAndGetPresetDir();
+  [[nodiscard]] static std::string CreateAndGetPluginDir();
+  [[nodiscard]] static std::string CreateAndGetCapturePath();
+  [[nodiscard]] static std::string CreateAndGetDumpPath();
+  [[nodiscard]] static std::string CreateAndGetOrbitAppDataPath();
+  [[nodiscard]] static std::string GetLogFilePathAndCreateDir();
+  [[nodiscard]] static std::string GetIconsPath();
 
 #ifdef __linux__
-  static std::string GetHome();
+  [[nodiscard]] static std::string GetHome();
 #endif
 
-  static std::string GetFileName(const std::string& a_FullName);
-  static std::string GetFileNameNoExt(const std::string& a_FullName);
-  static std::string StripExtension(const std::string& a_FullName);
-  static std::string GetExtension(const std::string& a_FullName);
-  static std::string GetDirectory(const std::string& a_FullName);
-  static std::string GetParentDirectory(std::string a_FullName);
-  static std::string JoinPath(const std::vector<std::string>& parts);
+  [[nodiscard]] static std::string GetFileName(const std::string& a_FullName);
+  [[nodiscard]] static std::string GetFileNameNoExt(const std::string& a_FullName);
```

Comment:
unused - remove?

Reply:
if abseil or filesystem has the equivalent, I think we should remove them yes

Reply:
I think this is obsolete, not only because it's unused, but because `StripExtension(GetFileName())` provides the same. Removed.